app/file.go

```diff
@@ -349,17 +346,20 @@ func (a *App) UploadFiles(teamId string, channelId string, userId string, files
 	return resStruct, nil
 }
 
-func (a *App) DoUploadFile(now time.Time, rawTeamId string, rawChannelId string, rawUserId string, rawFilename string, data []byte) (*model.FileInfo, *model.AppError) {
+func (a *App) DoUploadFile(now time.Time, rawTeamId string, rawChannelId string, rawUserId string, rawFilename string, file io.Reader) (*model.FileInfo, *model.AppError) {
 	filename := filepath.Base(rawFilename)
 	teamId := filepath.Base(rawTeamId)
 	channelId := filepath.Base(rawChannelId)
 	userId := filepath.Base(rawUserId)
 
-	info, err := model.GetInfoForBytes(filename, data)
+	info, err := model.GetInfoForBytes(filename, file)
 	if err != nil {
 		err.StatusCode = http.StatusBadRequest
 		return nil, err
 	}
+	if bio, ok := file.(io.Seeker); ok {
```

Comment:
If file is not an `io.Seeker`, of course this won't get executed and `file` will now be empty. That seems problematic, unless we can force `io.Seeker` onto all callers and mark it as such in the input parameters?